tests/unit/test_charm.py

```diff
@@ -2,8 +2,275 @@
 # See LICENSE file for licensing details.
 
 import unittest
+from unittest.mock import call, mock_open, patch
 
+from ops import testing
+from ops.model import ActiveStatus
 
-class TestCharm(unittest.TestCase):
-    def test_given_when_then(self):
+from charm import SrsLteCharm
+
+testing.SIMULATE_CAN_CONNECT = True
+
+
+class MockOpen:
+    def __init__(self, read_data: str = ""):
+        """Init."""
+        self.read_data = read_data
+        self.writen_data = None
+
+    def __enter__(self):
+        """Enter."""
+        return self
+
+    def read(self):
+        """Read."""
+        return self.read_data
+
+    def write(self, data: str):
+        """Write."""
+        self.writen_data = data
+
+    def __exit__(self, *args):
+        """Exit."""
         pass
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self) -> None:
+        self.harness = testing.Harness(SrsLteCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("os.mkdir")
+    @patch("shutil.copy")
+    @patch("shutil.rmtree")
+    @patch("subprocess.run")
+    def test_given_list_of_packages_to_install_when_install_then_apt_cache_is_updated(
+        self, patch_subprocess_run, _, __, ___, ____
+    ):
+
+        self.harness.charm.on.install.emit()
+
+        patch_subprocess_run.assert_any_call(
+            "sudo apt -qq update", shell=True, stdout=-1, encoding="utf-8"
+        )
+
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("os.mkdir")
+    @patch("shutil.copy")
+    @patch("shutil.rmtree")
+    @patch("subprocess.run")
+    def test_given_list_of_packages_to_install_when_install_then_apt_packages_are_installed(
+        self, patch_subprocess_run, _, __, ___, ____
+    ):
+        self.harness.charm.on.install.emit()
+
+        patch_subprocess_run.assert_any_call(
+            "sudo apt -y install git libzmq3-dev cmake build-essential libmbedtls-dev libboost-program-options-dev libsctp-dev libconfig++-dev libfftw3-dev net-tools",  # noqa: E501
```

Comment:
Could we fetch the list of packages from the charm (`self.harness.charm.APT_REQUIREMENTS`) of patch it?

Reply:
No, you want your unit tests to present what is expected of the code, not to depend on it. Let's say we replace the list in the code with 2 lists or whatever, we still expect at the end to call the `apt install ...`

Reply:
We do care about those specific packages being installed. If this list of package changes, we want the tests to also change. The test should validate that we install `git` for example.

Reply:
If that's the intention, then sure, we good. But in that case I'd say it's a configuration test rather than behavior test.